lib/vrl/compiler/src/type_def.rs

```diff
@@ -102,7 +102,9 @@ impl TypeDef {
                     coalesced_path: CoalescedPath::Reject,
                 },
             )
-            .unwrap_or(self.kind);
```

Comment:
I might pull this out of this PR since it's not related. Made an issue to track it since there are more places this impacts: https://github.com/vectordotdev/vector/issues/13460